src/coreComponents/physicsSolvers/wavePropagation/pygeosx/functions.py

```diff
@@ -0,0 +1,221 @@
+'''
+Created on 8/02/2021
+
+@author: macpro
+'''
+
+import math as m
+from functions import *
+import numpy as np
+from mesh import *
+from acquisition import *
+import pygeosx
+import sys
+from mpi4py import MPI
+from matplotlib import pyplot as plt
+from segy import *
+from print import *
+
+    
+    
+def callback(matrix, array):
+    pass
+    
+    
+    
+    
+def shot_simul(shot_list, dt):
+    problem = initialize()   
+    
+    do_shots(problem, shot_list, dt)
+    
+    
+       
+
+def initialize():
+    problem = pygeosx.reinit(sys.argv)
+    
+    try:
+        problem.get_group("Solvers/acousticSolver", None).register(callback)
+    except AttributeError:
+        pass
+
+    pygeosx.apply_initial_conditions()
+    
+    return problem
+
+
+
+def do_shots(problem, shot_list, dt):
+
+    #Get Acoustic group
+    acoustic_group  = problem.get_group("Solvers/acousticSolver")
+    
+    #Get Wrappers
+    src_pos_geosx   = acoustic_group.get_wrapper("sourceCoordinates").value()
+    src_pos_geosx.set_access_level(pygeosx.pylvarray.MODIFIABLE)
+    src_pos_geosx   = src_pos_geosx.to_numpy()
+    
+    rcv_pos_geosx_temp   = acoustic_group.get_wrapper("receiverCoordinates").value()
+    rcv_pos_geosx_temp.set_access_level(pygeosx.pylvarray.RESIZEABLE)
+    
+    pressure_geosx  = acoustic_group.get_wrapper("pressureNp1AtReceivers").value()
+    pressure_geosx  = pressure_geosx.to_numpy()
+    
+    pressure_nm1 = problem.get_wrapper("domain/MeshBodies/mesh/Level0/nodeManager/pressure_nm1").value()
+    pressure_nm1.set_access_level(pygeosx.pylvarray.MODIFIABLE)
+    pressure_nm1 = pressure_nm1.to_numpy()
+    
+    pressure_n = problem.get_wrapper("domain/MeshBodies/mesh/Level0/nodeManager/pressure_n").value()
+    pressure_n.set_access_level(pygeosx.pylvarray.MODIFIABLE)
+    pressure_n = pressure_n.to_numpy()
+    
+    pressure_np1 = problem.get_wrapper("domain/MeshBodies/mesh/Level0/nodeManager/pressure_np1").value()
+    pressure_np1.set_access_level(pygeosx.pylvarray.MODIFIABLE)
+    pressure_np1 = pressure_np1.to_numpy()
```

Comment:
I don't recommend storing numpy views in between calls to `pygeosx.run` because they will become invalid and you will if the underlying array gets resized, and you won't get any error message. Holding on to a wrapper is better since GEOSX rarely if ever deletes a wrapper.

Reply:
Hi @corbett5, thanks for your comments, I made the modifications to directly use the wrappers.